### PR TITLE
DSS SmokeTest: jmespath support

### PR DESCRIPTION
**Pull request by amarjandu**
adds support for testing `jmespath` subscriptions in test_smoketest.py
refactored test_prod_smoketest.py

### CODE REVIEW

**Comment by amarjandu** [comment]
`make smoketest` was also run locally to ensure it works correctly.

File: tests/test_smoketest.py [hunk 1]
```diff
@@ -52,25 +52,28 @@ def smoketest(self, starting_replica, checkout_bucket, test_bucket):
             'query': {'match': {'files.cell_suspension_json.biomaterial_core.biomaterial_id': biomaterial_id}}
         }
         tombstone_query = {"query": {"bool": {"must": [{"term": {"admin_deleted": "true"}}]}}}
-        queries = [bundle_query, tombstone_query]
+        tombstone_query_jmes = "event_type=='CREATE'"
+        queries = [(bundle_query, "elasticsearch"), (tombstone_query, "elasticsearch"),
+                   (tombstone_query_jmes, "jmespath")]
```
Comment (xbrianh) [inline tests/test_smoketest.py]: Why isn't there a bundle query for JMESPath?

File: tests/infra/base_smoketest.py [hunk 2]
```diff
@@ -150,6 +150,12 @@ def post_search_es(self, replica, es_query):
         """ post-search using es, returns post-search response """
         return run_for_json(f'{self.venv_bin}hca dss post-search  --es-query {es_query} --replica {replica.name}')
 
+    def put_subscription(self, replica, subscription_type, query, url):
```
Comment (xbrianh) [inline tests/infra/base_smoketest.py]: Consider inlining `subscription_put_jmespath` and `subscription_put_es` into this method.

File: tests/test_smoketest.py (hunk 1 above)
Comment (amarjandu) [reply tests/test_smoketest.py]: ah, it got lost; ill find it.

File: tests/infra/base_smoketest.py (hunk 2 above)
Comment (amarjandu) [reply tests/infra/base_smoketest.py]: ah thanks for this, reminded me i had to refactor the prod_smoketest 🙌

**Pull request merged**